src/bitcoind/interface.rs

```diff
@@ -89,64 +86,46 @@ impl BitcoinD {
             node_client,
             watchonly_client,
             cpfp_client,
+            try_limit: 1,
         })
     }
 
-    // Reasonably try to be robust to possible spurious communication error.
-    fn handle_error(&self, e: jsonrpc::Error, start: Instant) -> Result<(), BitcoindError> {
-        let now = Instant::now();
-
-        match e {
-            jsonrpc::Error::Transport(ref err) => {
-                log::error!("Transport error when talking to bitcoind: '{}'", err);
-
-                // This is *always* a simple_http::Error. Rule out the error that can
-                // not occur after startup (ie if we encounter them it must be startup
-                // and we better be failing quickly).
-                let any_err = err as &dyn Any;
-                if let Some(http_err) = any_err.downcast_ref::<HttpError>() {
-                    match http_err {
-                        HttpError::InvalidUrl { .. } => return Err(BitcoindError::Server(e)),
-                        // FIXME: allow it to be unreachable for a handful of seconds,
-                        // but not at startup!
-                        HttpError::SocketError(_) => return Err(BitcoindError::Server(e)),
-                        HttpError::HttpParseError => {
-                            // Weird. Try again once, just in case.
-                            if now.duration_since(start) > Duration::from_secs(1) {
-                                return Err(BitcoindError::Server(e));
+    /// Set the try limit field limiting the number of try a client is allowed to do for a request.
+    pub fn with_try_limit(mut self, retry_limit: usize) -> Self {
+        self.try_limit = retry_limit;
+        self
+    }
+
+    /// Code can not prevent Connection and I/O Errors to happen, in that case
+    /// the client can retry multiple time its request in the given limit.
```

Comment:
I find
> Code can not prevent Connection and I/O Errors to happen

confusing. How about:
```suggestion
    /// Wrapper to retry a request sent to bitcoind upon IO failure according to the configured number of retries.
```